pkg/sectigo/mock/mock.go

```diff
@@ -30,31 +31,84 @@ import (
 
 var mockServer *Server
 
+type Profile struct {
+	ID      string
+	Name    string
+	Balance int
+	Params  []string
+}
+
+var subjectParams = [4]string{
+	"organizationName",
+	"localityName",
+	"stateOrProvinceName",
+	"countryName",
+}
+
+var nameParams = [3]string{
+	"commonName",
+	"dNSName",
+	"pkcs12Password",
+}
+
+// All the profiles that the mock knows about.
+var validProfiles = []Profile{
+	{
+		ID:      sectigo.ProfileIDCipherTraceEE,
+		Name:    sectigo.ProfileCipherTraceEE,
+		Balance: 50,
+		Params:  nameParams[:],
+	},
+	{
+		ID:      sectigo.ProfileIDCipherTraceEndEntityCertificate,
+		Name:    sectigo.ProfileCipherTraceEndEntityCertificate,
+		Balance: 100,
+		Params:  append(nameParams[:], subjectParams[:]...),
+	},
+}
+
 // Server helps verify that the success paths of the Sectigo API calls are working,
 // while also providing a way for tests to inject handlers to test the errors that
 // might be returned from the Sectigo API.
 type Server struct {
-	server   *httptest.Server
-	router   *gin.Engine
-	handlers sync.Map
-	calls    sync.Map
+	server          *httptest.Server
+	router          *gin.Engine
+	handlers        sync.Map
+	calls           sync.Map
+	profileLookup   map[string]Profile
+	authorityLookup map[int]Profile
 }
 
 // Get returns the current mock server.
 func Get() *Server {
 	return mockServer
 }
 
-// Start initializes a new server which mocks the Sectigo REST API. By default, it sets up
+// Start() initializes a new server which mocks the Sectigo REST API. By default, it sets up
 // HTTP handlers which return 200 OK responses with mocked data, but custom handlers can
 // be passed in by tests to test specific error paths. Note that this function modifies
 // the externally used baseURL for the Sectigo endpoint, and the caller must call Stop()
-// to close the server and undo the mock when done.
-func Start() error {
+// to close the server and undo the mock when done. Start() takes one paramter, which is the
+// name of the sectigo profile that the mock server uses to verify request parameters in
+// createSingleCertBatch.
+func Start(profile string) error {
 	gin.SetMode(gin.TestMode)
 
 	mockServer = &Server{
-		router: gin.New(),
+		router:          gin.New(),
+		profileLookup:   make(map[string]Profile),
+		authorityLookup: make(map[int]Profile),
+	}
+
+	// TODO: We are only loading one authority so that we can get consistent behavior
+	// from the tests.
+	authority := 1
+	for _, p := range validProfiles {
+		mockServer.profileLookup[p.ID] = p
+		if p.Name == profile {
+			mockServer.authorityLookup[authority] = p
+		}
+		authority++
```

Comment:
This got a little hairy, originally I wanted to have the mock know about both authorities but I had forgotten that the cert manager code is just taking the first authority available and the profile ID from that, which made the tests unreliable. So, now `Start()` takes in a profile name and only loads one authority into the lookup map, which allows the tests to be able to configure which profile they want to test against.

Reply:
That makes sense and is closer to how Sectigo behaves anyway.

Reply:
So to answer my earlier question, if empty string is passed in, then `profileLookup` and `authorityLookup` will be empty, therefore the mock will always return 404s on those endpoints?

Reply:
Exactly